lib/entropy/dep.py

```diff
@@ -644,7 +644,8 @@ def entropy_compare_package_tags(tag_a, tag_b):
         zero if tag_a == tag_b.
     rtype: int
     """
-    return const_cmp(tag_a, tag_b)
+    return const_cmp(long(''.join(str(x) for x in map(ord,tag_a))),
```

Comment:
str() is ascii for Python 2.x, which is bad. Never use str() in any case.
Also, this code is way too complicated, what does it do? Please add comments.
Also 2, the function docstring would be then out of sync.

I argue that we should only tag kernel tags then, and match something like X.Y.Z-NAME strings. Regexp matching would be probably a bit slow though.

Reply:
why just kernels? we use tags also for other packages as well (apache, etc.) i don't see why we should do handle it as a specific case.

I would discourage use of regexes, are slow, error-prone, hard to maint and easy to break.